### ISSUE TITLE
Different results in 1.6 and 1.7 versions

### ISSUE THREAD

**Issue by Aljaksandr** [searchivarius/nmslib#286]
When querying in index with assigned to each data line specific id, the output in 1.7 version is different from the output in 1.6 version. In version 1.7 query returns just an index of found line. The sample of code:

```
import nmslib
import numpy

# create a random matrix to index
number = 100
data = numpy.random.randn(number, 10).astype(numpy.float32)
idxs = numpy.arange(0, number) + 123

# initialize a new index, using a HNSW index on Cosine Similarity
index = nmslib.init(method='hnsw', space='cosinesimil')
index.addDataPointBatch(data, idxs)
index.createIndex({'post': 2}, print_progress=True)

# query for the nearest neighbours of the first datapoint
ids, distances = index.knnQuery(data[0], k=1)

print(ids, distances)
```
It returns:
 [123] [ 0.] - in 1.6 version
 [0] [ 0.] - in 1.7 version

How to fix this issue in 1.7 ver?

**Comment by Aljaksandr** [comment]
Pushing data with
```
for (id, emb) in zip(idxs, data):
    index.addDataPoint(id, emb)
```
works correctly. It returns [123] [ 0.]

**Comment by searchivarius** [comment]
@Aljaksandr so IDs are not stored when you add addDataPoint**Batch**?

**Comment by Aljaksandr** [comment]
@searchivarius I suppose IDs aren't stored via addDataPointBatch

**Comment by searchivarius** [comment]
@Aljaksandr it's certainly possible, but the code to store them is there. It can be of course broken, I will soon investigated this, thanks a lot for reporting!

**Comment by searchivarius** [comment]
Hi @Aljaksandr thank you for reporting. The recent fix by @benfred seem to resolve the issue. The bug-fix is available in 1.7.1 (pip can install this version).

**Comment by Aljaksandr** [comment]
Thanks a lot for a fast fix. )
